autoPyTorch/pipeline/components/training/trainer/base_trainer_choice.py

```diff
@@ -337,6 +343,15 @@ def _fit(self, X: Dict[str, Any], y: Any = None, **kwargs: Any) -> torch.nn.Modu
 
             torch.cuda.empty_cache()
 
+        if self.choice.use_swa:
+            # update batch norm statistics
+            swa_utils.update_bn(X['train_data_loader'], self.choice.swa_model.double())
```

Comment:
and why casting it here to double?

Reply:
I think the data in the loader is of double precision and hence I was getting an error because the model was in float